### PR TITLE
refactor: Add JSON attributes to MessageEnvelope fields

### LINKED ISSUE #169: [EdgeX 3.0] Add JSON attributes to MessageEnvelope
The `MessageEnvelope` is missing the proper JSON attributes on the fields

### PR DESCRIPTION
**Pull request by lenny-intel**
Not a breaking change as original thought.

closes #169

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-messaging/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?) **Unit Test suffice**
- [ ] I have opened a PR for the related docs change (if not, why?) **N/A**
  <link to docs PR>

### CODE REVIEW

File: pkg/types/message_envelope.go [hunk 1]
```diff
@@ -36,29 +37,29 @@ const (
 
 // MessageEnvelope is the data structure for messages. It wraps the generic message payload with attributes.
 type MessageEnvelope struct {
+	common.Versionable
+	// RequestID is an object id to identify the request.
```
Comment (jim-wang-intel) [inline pkg/types/message_envelope.go:41]: this comment doc should be in line 46 below

File: pkg/types/message_envelope.go [hunk 2]
```diff
@@ -36,29 +37,29 @@ const (
...
 	// ReceivedTopic is the topic that the message was received on.
-	ReceivedTopic string
+	ReceivedTopic string `json:"receivedTopic"`
 	// CorrelationID is an object id to identify the envelope.
-	CorrelationID string
+	CorrelationID string `json:"correlationID"`
 	// ApiVersion shows the API version in message envelope.
```
Comment (jim-wang-intel) [inline pkg/types/message_envelope.go:46]: this comment doc should be removed?

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

File: pkg/types/message_envelope.go (hunk 2 above)
Comment (lenny-intel) [reply pkg/types/message_envelope.go]: fixed

File: pkg/types/message_envelope.go (hunk 1 above)
Comment (lenny-intel) [reply pkg/types/message_envelope.go]: fixed

**Review by jim-wang-intel — DISMISSED** [review]
LGTM

**Review by jim-wang-intel — APPROVED** [review]

**Comment by cloudxxx8** [comment]
this should be considered as a breaking change because the JSON payload field names are changed.
if the users made a service subscribing the message payload and it's case sensitive to parsing JSON, it would be a problem.

**Comment by lenny-intel** [comment]
Actually, per my testing, as long as the name stays the same the JSON un-marshal handles the difference in case in the names.

**Pull request merged**